application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/script/LicensorScriptService.java

```diff
@@ -237,4 +249,22 @@ public Long getUserCount()
             return null;
         }
     }
+
+    /**
+     * @param documentReference is the document for which is searching for associated sheets with missing license
+     * @param action represents the action made
+     * @return a list of DocumentReference (sheets) that are missing license for a specific documentReference and action
+     */
+    public List<DocumentReference> getUnlicensedSheets(DocumentReference documentReference, String action)
+    {
+        try {
+            DocumentModelBridge documentModelBridge = documentAccessBridge.getDocument(documentReference);
+            List<DocumentReference> sheetsUsedByDocument = sheetManager.getSheets(documentModelBridge, action);
+            return sheetsUsedByDocument.stream()
+                    .filter(sheet -> !this.hasLicensureForEntity(sheet))
+                    .collect(Collectors.toList());
+        } catch (Exception e) {
+            return null;
```

Comment:
We could also log an error here so it's easier to debug the problems that may occur.